fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BuildAggForRandomDistributedTable.java

```diff
@@ -0,0 +1,206 @@
+                    break;
+                }
+            }
+            if (col.isKey()) {
+                groupByExpressions.add(slot);
+                outputExpressions.add(slot);
+            } else {
+                AggregateType aggregateType = col.getAggregationType();
+                AggregateFunction function;
+                if (aggregateType == AggregateType.SUM) {
+                    function = new Sum(slot);
+                } else if (aggregateType == AggregateType.MAX) {
+                    function = new Max(slot);
+                } else if (aggregateType == AggregateType.MIN) {
+                    function = new Min(slot);
+                } else {
+                    // do not rewrite
+                    return logicalPlan;
+                }
+                Alias alias = new Alias(exprId, function, col.getName());
+                outputExpressions.add(alias);
+            }
+        }
+        LogicalAggregate<LogicalOlapScan> aggregate = new LogicalAggregate<>(groupByExpressions, outputExpressions,
+                olapScan);
+        return logicalPlan.withChildren(aggregate);
+    }
+
+    /**
+     * if the agg type of AggregateFunction is as same as the agg type of column, DO NOT need to rewrite
+     *
+     * @param function agg function to check
+     * @return true if agg type match
+     */
+    private boolean aggTypeMatch(AggregateFunction function) {
+        List<Expression> children = function.children();
+        String functionName = function.getName();
+        if (function.getName().equalsIgnoreCase("count")) {
+            Count count = (Count) function;
+            // do not rewrite for count distinct for key column
+            if (count.isDistinct()) {
+                return children.stream().allMatch(this::isKeyOrConstantExpr);
+            }
+            if (count.isStar()) {
+                return false;
+            }
+        }
+        return children.stream().allMatch(child -> aggTypeMatch(functionName, child));
+    }
+
+    /**
+     * check if the agg type of functionCall match the agg type of column
+     *
+     * @param functionName the functionName of functionCall
+     * @param expression expr to check
+     * @return true if agg type match
+     */
+    private boolean aggTypeMatch(String functionName, Expression expression) {
+        if (expression.children().isEmpty()) {
+            if (expression instanceof SlotReference && ((SlotReference) expression).getColumn().isPresent()) {
+                Column col = ((SlotReference) expression).getColumn().get();
+                if (col.isKey()) {
+                    return functionName.equalsIgnoreCase("max") || functionName.equalsIgnoreCase("min");
+                }
+                if (col.isAggregated()) {
+                    AggregateType aggType = col.getAggregationType();
+                    // agg type not mach
+                    return (aggType == AggregateType.SUM || aggType == AggregateType.MAX
```

Comment:
we have to deal with other agg types, including HLL_UNION, BITMAP_UNION, QUANTILE_UNION and GENERIC(agg_state related)

Reply:
done